File: Mündlich/SEW/SEW_4AHITM/Homework/excercise-4-geometry/Geometry/Geometry/GeometryObjects/Shapes/Point.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GeometryApp
{
    internal struct Point : IPoint
    {
        public Point(int x, int y)
        {
            this.x = x;
            this.y = y;
        }

        public int x {  get; set; } 
        public int y { get; set; }

        public static bool operator ==(Point a, Point b)
        {
            if (a.y == b.y && a.x == b.x) return true;
            
            return false;
        }

        public static bool operator !=(Point a, Point b)
        {
            if (a.y == b.y && a.x == b.x) return false;

            return true;
        }

        public override string ToString()
        {
            return $"X: {x} Y: {y}";
        }
    }
}

File: Mündlich/SEW/SEW_2AHIT/Java/School/Classes/src/com/company/Point.java
package com.company;

public class Point
{
    double xCoordinate;
    double yCoordinate;

    Point()
    {
        xCoordinate = 0;
        yCoordinate = 0;
    }

    Point(double xCoordinate, double yCoordinate)
    {
        this.xCoordinate = xCoordinate;
        this.yCoordinate = yCoordinate;
    }

    public void print()
    {
        System.out.println("X: " + this.xCoordinate);
        System.out.println("Y: " + this.yCoordinate);
    }

    @Override
    public String toString()
    {
        return "Point{" +
                "xCoordinate=" + xCoordinate +
                ", yCoordinate=" + yCoordinate +
                '}';
    }
}
